app/src/org/commcare/android/database/user/models/SessionStateDescriptor.java

```diff
@@ -91,7 +91,7 @@ private String createSessionDescriptor(CommCareSession session) {
         //TODO: Serialize into something more useful. I dunno. JSON/XML/Something
         StringBuilder descriptor = new StringBuilder();
         for (StackFrameStep step : session.getFrame().getSteps()) {
-            descriptor.append(step.getType());
+            descriptor.append(step.getType()).append(" ");
             if (SessionFrame.STATE_COMMAND_ID.equals(step.getType())) {
                 descriptor.append(step.getId());
```

Comment:
I think the one at the end handles for all the afterwards ones, but this is in between the key and the value